low-level/sensors/impl/generic/node_hw.py

```diff
@@ -636,109 +689,88 @@ def _run_ipmitool_subcommand(self, subcommand, grep_args=None, out_file=subproce
+            # If error detected for lan/lanplus interface,
+            # fallback to KCS interface.
+            if self.active_bmc_if in BMCInterface.LAN_IF.value:
+                logger.warning(
+                    f"BMC is unreachable through {self.active_bmc_if}"
+                    "interface, ipmitool fallback to system interface if local"
+                    "server is being monitored."
+                    )
+                self.active_bmc_if = system
+                store.put(self.active_bmc_if, active_bmc_if)
+
+    def get_channel_alert(self, alert, IF_name, err=""):
+        """create BMC interface alert json msg."""
 
-        resource_id = channel_info.get('Channel Medium Type')
-        severity_reader = SeverityReader()
+        specific_info = {}
+        alert_type = alert.alert
+        severity = SeverityReader().map_severity(alert_type)
+        channel_info = self.CHANNEL_INFO
 
+        if IF_name in BMCInterface.LAN_IF.value:
+            resource_type = "node:bmc:interface:rmcp"
+            specific_info["bmc_user"] = self._bmc_user
+            if alert_type == "fault":
+                channel_info = self.LAN_CHANNEL_INFO
+        else:
+            resource_type = "node:bmc:interface:kcs"
+            if alert_type == "fault":
+                channel_info = self.KCS_CHANNEL_INFO
         info = {
                 "resource_type": resource_type,
-                "resource_id": resource_id,
-                "event_time": epoch_time,
-                "description": channel_status
-            }
-
-        specific_info = {
-                "channel info": channel_info
+                "resource_id": IF_name,
+                "event_time": str(int(time.time())),
+                "description": alert.description.format(IF_name, err),
+                "impact": alert.impact.format(IF_name),
+                "recommendation": alert.recommendation.format(IF_name)
             }
-
-        if channel_info.get('Channel Medium Type'):
-            if_type = channel_info.get('Channel Medium Type')
-            if if_type == "802.3 LAN":
-                specific_info["bmc_lan_user"] = self._bmc_user
-
-        if alert_type is not None:
-            severity = severity_reader.map_severity(alert_type)
-            self._send_json_msg(resource_type, alert_type, severity, info, specific_info)
-
-    def _check_kcs_if_alert(self,err,retcode):
-
-        if retcode != 0:
-            # TODO: search for error codes which ipmitool returns in case of channel error
-            if err in self.ipmi_client.KCS_ERRS:
-                self.kcs_interface_alert = True
-
-    def _check_lan_if_alert(self,err,retcode):
-
-        if retcode !=0:
-            if err in self.ipmi_client.RMCP_ERRS:
-                self.lan_interface_alert = True
+        specific_info["channel info"] = channel_info
+        # Update cache
+        data = ""
+        key = ""
+        if IF_name == system:
```

Comment:
again all such code can be merged if we dont maintain separate vars for different interfaces, unless they are unavoidable

Reply:
Instead of using seperate vars for each interface used one var lan_fault for - lan and lanplus
and one for system.

Reply:
yes, use 2 vars